File: networking/shout/shout.py
import socket

s = socket.socket(family=socket.AF_INET, type=socket.SOCK_DGRAM)
s.bind(('localhost', 1337))

while True:
    msg, sender = s.recvfrom(1024)
    print(f'{sender}: {msg}')
    s.sendto(msg.decode().upper().encode(), sender)

File: rustimpls/networking/src/bin/shout.rs
use std::net::UdpSocket;

fn main() -> std::io::Result<()> {
    let socket = UdpSocket::bind("127.0.0.1:5050")?;
    println!("listening @ localhost:5050");
    
    loop {
        let mut buf = [0; 1024];
        let (amt, src) = socket.recv_from(&mut buf)?;

        let msg = buf.to_ascii_uppercase();
        socket.send_to(&msg, &src)?;
        println!("Received {} bytes from {}", amt, src);
    }
}